### PR TITLE
avoid DoesNotExist when querying a machine

### PR DESCRIPTION
**Pull request by pchristos**
catch DoesNotExist in case machine object cannot be found in mongo.
update key_assoc only if machine is stored locally

---
This change is [Reviewable](https://reviewable.io/reviews/mistio/mist.io/742)

### CODE REVIEW

File: src/mist/io/methods.py [hunk 1]
```diff
@@ -2537,11 +2537,15 @@ def _machine_action(user, cloud_id, machine_id, action, plan_id=None, name=None)
...
-                                              machine_id=machine_id)
-                for key_assoc in machine.key_associations:
-                    key_assoc.port = port
-                machine.save()
+                try:
+                    machine = Machine.objects.get(cloud=cloud,
+                                                  machine_id=machine_id)
+                except DoesNotExist:
+                    pass
+                else:
```
Comment (mgogoulos) [inline src/mist/io/methods.py]: where is this else refering to? I don't see an if...ahead

File: src/mist/io/methods.py [hunk 2]
```diff
@@ -2598,12 +2602,16 @@ def _machine_action(user, cloud_id, machine_id, action, plan_id=None, name=None)
...
 
+                    try:
+                        machine = Machine.objects.get(cloud=cloud,
+                                                      machine_id=machine_id)
+                    except DoesNotExist:
+                        pass
+                    else:
+                        for key_assoc in machine.key_associations:
+                            key_assoc.port = port
+                        machine.save()
```
Comment (mgogoulos) [inline src/mist/io/methods.py]: on the try/except a few lines before, if machine is not found, this machine.save() will break

File: src/mist/io/methods.py (hunk 1 above)
Comment (ptzianos) [inline src/mist/io/methods.py]: the else block will be executed if no exception is thrown

**Comment by ptzianos** [comment]
@pchristos can you add a couple of api tests for verifying this code?

**Comment by pchristos** [comment]
The code has been tested both locally, as well as via a new API test. It checks out

**Comment by ptzianos** [comment]
Run tests

**Pull request merged**